routing.go

```diff
@@ -93,39 +97,58 @@ func (dht *IpfsDHT) GetValue(ctx context.Context, key string) (_ []byte, err err
 	ctx, cancel := context.WithTimeout(ctx, time.Minute)
 	defer cancel()
 
-	vals, err := dht.GetValues(ctx, key, 16)
-	if err != nil {
+	var cfg ropts.Options
+	if err := cfg.Apply(opts...); err != nil {
 		return nil, err
 	}
 
-	recs := make([][]byte, 0, len(vals))
-	for _, v := range vals {
-		if v.Val != nil {
-			recs = append(recs, v.Val)
-		}
-	}
-	if len(recs) == 0 {
-		return nil, routing.ErrNotFound
-	}
-
-	i, err := dht.Selector.BestRecord(key, recs)
+	results, err := dht.getValues(ctx, key, &cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	best := recs[i]
-	log.Debugf("GetValue %v %v", key, best)
-	if best == nil {
-		log.Errorf("GetValue yielded correct record with nil value.")
-		return nil, routing.ErrNotFound
+	var outdatedPeers, currentPeers []peer.ID
+
+	var best []byte
+	for result := range results {
+		switch {
+		case result.val == nil:
+			outdatedPeers = append(outdatedPeers, result.from)
+		case best == nil:
+			best = result.val
+			fallthrough
+		case bytes.Equal(result.val, best):
+			currentPeers = append(currentPeers, result.from)
+		default:
+			i, err := dht.Selector.BestRecord(key, [][]byte{best, result.val})
```

Comment:
Seems like we just want a simple record comparer now, instead of a 'select best' function

Reply:
It might be simpler to find the index of the best record using `BestRecord()`, then for each record whose value is not the same as `best`, send the corresponding peer an update.
It also occurs to me that it might make more sense for the [Selector](https://github.com/libp2p/go-libp2p-record/blob/master/selection.go#L9) interface to look something like the [Sort](https://golang.org/pkg/sort/) interface, ie compare two values at a time rather than operating over the whole list.